src/backend/dx11/src/lib.rs

```diff
@@ -1730,21 +961,56 @@ pub struct UnboundBuffer {
     requirements: memory::Requirements,
 }
 
+#[derive(Clone)]
+pub enum InternalBuffer {
+    Coherent(ComPtr<d3d11::ID3D11Buffer>),
+    NonCoherent {
+        device: ComPtr<d3d11::ID3D11Buffer>,
```

Comment:
why do we need a device local counterpart for the non-coherent buffer type here?